### PR TITLE
Fixed #27355 -- Added “Create extension using migrations”

### PR DESCRIPTION
**Pull request by ntuckovic**
Additional explanation for PostreSQL extensions installation via migrations, with hstore extension as an example.

### CODE REVIEW

File: docs/ref/contrib/postgres/operations.txt [hunk 1]
```diff
@@ -59,3 +59,34 @@ the ``django.contrib.postgres.operations`` module.
 .. class:: UnaccentExtension()
 
     Installs the ``unaccent`` extension.
+
+``Create extension using migrations``
+=====================================
+
+You can create a PostgreSQL Extension in your database through the generated
```
Comment (mr-bo-jangles) [inline docs/ref/contrib/postgres/operations.txt]: I would change this portion of the sentence to "in your database using a migration file. The following"

File: docs/ref/contrib/postgres/operations.txt [hunk 2]
```diff
@@ -59,3 +59,34 @@ the ``django.contrib.postgres.operations`` module.
...
+                HStoreExtension(),
+                ...
+            ]
+
+   Creating the extension requires a database user with superuser
+   privileges. If the Django database user doesn't have superuser
+   privileges, you'll have to create the extension outside of Django
+   migrations with a user that has the appropriate privileges.
+
+   For hstore extension, connect to your Django database and run the query:
```
Comment (mr-bo-jangles) [inline docs/ref/contrib/postgres/operations.txt]: I'd change this to "For example, doing this manually with the hstore extension, you would connect to your Django database and run the following query:"

File: docs/ref/contrib/postgres/operations.txt (hunk 1 above)
Comment (ntuckovic) [inline docs/ref/contrib/postgres/operations.txt]: Your suggestion looks much more abstract, and I like it, will do.

File: docs/ref/contrib/postgres/operations.txt (hunk 2 above)
Comment (ntuckovic) [inline docs/ref/contrib/postgres/operations.txt]: I'm gonna trust your "native english speaking skills" 😄 . Will do it.

File: docs/ref/contrib/postgres/operations.txt [hunk 3]
```diff
@@ -59,3 +59,35 @@ the ``django.contrib.postgres.operations`` module.
 .. class:: UnaccentExtension()
 
     Installs the ``unaccent`` extension.
+
+``Create extension using migrations``
+=====================================
+
+You can create a PostgreSQL Extension in your database using a migration file.
+The following example will cover creating of hstore extension,
```
Comment (reinout) [inline docs/ref/contrib/postgres/operations.txt]: Spelling: "example will cover creating of hstore extension" => "example covers creating an hstore extension"

File: docs/ref/contrib/postgres/operations.txt [hunk 4]
```diff
@@ -59,3 +59,35 @@ the ``django.contrib.postgres.operations`` module.
...
+but same usage principles apply for all other extensions, mentioned above.
+
+1. Add ``'django.contrib.postgres'`` in your :setting:`INSTALLED_APPS`.
```
Comment (reinout) [inline docs/ref/contrib/postgres/operations.txt]: There is a big difference in length of the first and second point. Better not to use a numbered list. Perhaps do the first point as a separate paragraph like this?

    Prerequisite: ``'django.contrib.postgres'`` must be in your :setting:`INSTALLED_APPS` setting.

File: docs/ref/contrib/postgres/operations.txt [hunk 5]
```diff
@@ -59,3 +59,35 @@ the ``django.contrib.postgres.operations`` module.
...
+2. Setup the hstore extension in PostgreSQL before the first ``CreateModel``
```
Comment (reinout) [inline docs/ref/contrib/postgres/operations.txt]: Spelling: "Setup" => "Set up" (two words).

File: docs/ref/contrib/postgres/operations.txt [hunk 6]
```diff
@@ -59,3 +59,35 @@ the ``django.contrib.postgres.operations`` module.
...
+   For example::
+
+        from django.contrib.postgres.operations import HStoreExtension
+
+        class Migration(migrations.Migration):
+            ...
+
+            operations = [
+                HStoreExtension(),
+                ...
```
Comment (reinout) [inline docs/ref/contrib/postgres/operations.txt]: These dots suggests there is more you should do. Is that mandatory? Wouldn't your example (without the dots) be good enough as a single migration?

(Note: I'm not an expert on postgres extensions, so I might be wrong).

File: docs/ref/contrib/postgres/operations.txt [hunk 7]
```diff
@@ -59,3 +59,35 @@ the ``django.contrib.postgres.operations`` module.
...
+            ]
+
+   Creating the extension requires a database user with superuser
```
Comment (reinout) [inline docs/ref/contrib/postgres/operations.txt]: Good warning! Great that you thought of it.

File: docs/ref/contrib/postgres/operations.txt [hunk 8]
```diff
@@ -59,3 +59,35 @@ the ``django.contrib.postgres.operations`` module.
...
+   privileges. If the Django database user doesn't have superuser
+   privileges, you'll have to create the extension outside of Django
+   migrations with a user that has the appropriate privileges.
+
+   For example, doing this manually with the hstore extension,
```
Comment (reinout) [inline docs/ref/contrib/postgres/operations.txt]: Spelling: "with" => "for"

**Comment by timgraham** [comment]
Edited and merged in 4de8aaf7ffc91b91cbb70e9db406abe9b160575a, thanks!

**Pull request closed without merging**